sdk/core/Azure.Core/src/Shared/DiagnosticScope.cs

```diff
@@ -528,6 +514,22 @@ public void SetTraceContext(string traceparent, string? tracestate)
                 _tracestate = tracestate;
             }
 
+            private void AddObjectTag(string name, object value)
+            {
+#if NETCOREAPP2_1
+                _currentActivity?.AddTag(name, value.ToString());
```

Comment:
Does a null check for `value` make sense here, or is it guaranteed to be not null?

Reply:
good catch!
this is guaranteed to be not null here (we back off for null values before calling into this method).